### PR TITLE
#4067: Add support for kinesis retention hours

### LINKED ISSUE #4067: Add support for retention hours in Kinesis
Hello,

Thanks for this great library! I would like to contribute to moto, and I noticed that the methods to interact with the retention_hours present in Kinesis are not yet implement(specifically `increase_stream_retention_period` and `decrease_stream_retention_period`. I would like to work on implementing these functions.

Thanks!

### PR DESCRIPTION
**Pull request by goodhamgupta**
Fixes #4067.

This PR aims to add support for the functions `increase_stream_retention_period` and `decrease_stream_retention_period` in Kinesis.

### CODE REVIEW

File: tests/test_kinesis/test_kinesis.py [hunk 1]
```diff
@@ -479,6 +481,65 @@ def test_get_records_from_empty_stream_at_timestamp():
...
+@mock_kinesis
+def test_invalid_increase_stream_retention_period():
+    conn = boto3.client("kinesis", region_name="us-west-2")
+    stream_name = "my_stream"
+    conn.create_stream(StreamName=stream_name, ShardCount=1)
+
+    conn.increase_stream_retention_period(
+        StreamName=stream_name, RetentionPeriodHours=30
+    )
+    conn.increase_stream_retention_period.when.called_with(
```
Comment (bblommers) [inline tests/test_kinesis/test_kinesis.py:507]: Negative tests should use pytest.raises, to validate that the error is raised in the correct format:
```
with pytest.raises(ClientError) as ex:
        conn.increase_stream_retention_period(..)
ex.value.response["Error"]["Code"].should.equal("...")
ex.value.response["Error"]["Code"].should.equal("...")
```

File: tests/test_kinesis/test_kinesis.py [hunk 2]
```diff
@@ -479,6 +481,65 @@ def test_get_records_from_empty_stream_at_timestamp():
...
+@mock_kinesis
+def test_invalid_decrease_stream_retention_period():
+    conn = boto3.client("kinesis", region_name="us-west-2")
+    stream_name = "decrease_stream"
+    conn.create_stream(StreamName=stream_name, ShardCount=1)
+
+    conn.increase_stream_retention_period(
+        StreamName=stream_name, RetentionPeriodHours=30
+    )
+    conn.decrease_stream_retention_period.when.called_with(
```
Comment (bblommers) [inline tests/test_kinesis/test_kinesis.py:538]: Same here

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @goodhamgupta, thanks for the PR! Please see my comments

**Review by bblommers — APPROVED** [review]
LGTM!

**Comment by goodhamgupta** [comment]
Thanks for your kind comments and response @bblommers! Looking forward to contributing more to moto soon 😄

**Pull request merged**